app/models/project_folder.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class ProjectFolder < ApplicationRecord
+  include SearchableModel
+  include SearchableByNameModel
+
+  validates :name,
+            length: { minimum: Constants::NAME_MIN_LENGTH,
+                      maximum: Constants::NAME_MAX_LENGTH },
+            uniqueness: { scope: :team_id, case_sensitive: false }
+
+  before_validation :inherit_team_from_parent_folder, if: -> { parent_folder.present? }
+
+  belongs_to :team, inverse_of: :project_folders, touch: true
```

Comment:
Why do we want to touch the team on each project_folder change?

Reply:
To invalidate caches